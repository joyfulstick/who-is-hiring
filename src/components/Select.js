import React from 'react'
import LabelStyle from '../styles/LabelStyle'

const Select = ({ name, value, handleChange, options }) => (
  <LabelStyle>
    {name}
    <select name={name} value={value} onChange={handleChange}>
      {options.map(o => (
        <option key={o} value={o}>
          {o}
        </option>
      ))}
    </select>
  </LabelStyle>
)

export default Select
