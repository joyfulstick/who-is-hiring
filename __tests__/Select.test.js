import { shallow } from 'enzyme'
import React from 'react'
import Select from '../src/components/Select'

describe('<Select />', () => {
  it('renders correctly - type checkbox', () => {
    const wrapper = shallow(
      <Select
        name="position"
        value="all"
        checked={true}
        options={['all', 'engineer', 'data scientist', 'designer', 'manager']}
        handleChange={jest.fn()}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
