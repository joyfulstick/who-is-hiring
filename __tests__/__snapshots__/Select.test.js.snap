// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly - type checkbox 1`] = `
<styled.label>
  <span>
    position
  </span>
  <select
    name="position"
    onChange={[MockFunction]}
    value="all"
  >
    <option
      key="all"
      value="all"
    >
      all
    </option>
    <option
      key="engineer"
      value="engineer"
    >
      engineer
    </option>
    <option
      key="data scientist"
      value="data scientist"
    >
      data scientist
    </option>
    <option
      key="designer"
      value="designer"
    >
      designer
    </option>
    <option
      key="manager"
      value="manager"
    >
      manager
    </option>
  </select>
</styled.label>
`;
